src/libpmemfile-posix/dir.c

```diff
@@ -1242,23 +1226,46 @@ _pmemfile_rmdirat(PMEMfilepool *pfp, struct pmemfile_vinode *dir,
 		goto vparent_end;
 	}
 
+	dirent = NULL;
```

Comment:
To prevent future me (or any developer) from accidentally using it after unlock.